docs/architecture/blockchain.md

```diff
@@ -0,0 +1,70 @@
+# Blockchain
+
+The Miden blockchain protocol describes how the [state](state.md) progresses. In Miden, `Block`s are containers that aggregate account state changes and their proofs, together with created and consumed notes. Each `Block` is accompanied by a corresponding proof that attests to the correctness of all state transitions it contains.
+
+`Block`s represent the delta of the global [state](state.md) between two time periods. The current global state can be derived by applying all the blocks to the genesis state.
```

Comment:
This paragraph seems a bit redundant with the previous one. There is probably a good way to combine them.